lib/modules/manager/gradle/artifacts.ts

```diff
@@ -149,7 +149,11 @@ export async function updateArtifacts({
       .map(quote)
       .join(' ')}`;
 
-    if (config.isLockFileMaintenance || isGcvPropsFile(packageFileName)) {
+    if (
+      config.isLockFileMaintenance ||
+      !updatedDeps.length ||
```

Comment:
🤔 can't this cause unwanted side effects? 🤔

Reply:
Shall we make the gradle wrapper use case invocation more explicit rather than implicit?

Reply:
sounds good

Reply:
> 🤔 can't this cause unwanted side effects? 🤔

I can't think of any:

- Whenever `updateArtifacts()` is called after regular dep updates, `updatedDeps` isn't emtpy.
- In the `else` clause, `--update-locks` needs a non-empty `updatedDeps` to compose a valid argument. So checking `!updatedDeps.length` in the `if` clause seemed safe for me.

The reason I went this route is to prevent fiddling with the `config.isLockFileMaintenance` option to get to `--write-locks`.

@rarkins are you referring to the docs or code changes?

Reply:
OK, I'm fine with that. Thanks